docs/src/config/model.md

```diff
@@ -54,7 +54,19 @@ scale based on the bounding box of the computational domain.
             "Radius": float
         },
         ...
-    ]
+    ],
+    "Adaptation":
```

Comment:
Part of the rationale for the substruct was that I had been thinking of Adaptation as a subtype of Refinement. If the idea is that Refinement becomes Adaptation however, then this could change. A lot of these options would then need to be prefixed with `Adaptive` or similar. I thought the substruct was more encapsulating though, hence this choice.

The main advantage of moving to Advanced options is not having to write a doc string for an option, but given I have I'm not sure the extra information is worth removing. Independent of that issue though:
* I could definitely see `"SaveStep"` equal to 0, as that amounts to just overwriting the postpro directory every time a new solve completes, without saving off the substeps. If a user didn't care about storing the intermediate results (or cared about disk space), they might set that option. This could be changed to a boolean instead, of storing results or not, but I originally picked `"SaveStep"` for consistency with other saving off options.
* `"MaximumImbalance"` could be argued as advanced, as adjusting that from 1 ought to be very unusual. 
* `"WriteSerialMesh"` is similar in spirit to the `"SaveStep"` option, if a user didn't care for the extra storage. It's pretty self explanatory though, so again can be made advanced. This option is mainly helpful for cases where a user thinks they might conceivably want to restart on a larger machine, or generate an adapted mesh for use in a parametric sweep.
* `"UseCoarsening"` could also be argued as advanced, given the default to false, and that any reasonable choice of `"UpdateFraction"` makes it practically unneeded. This argument can be extended to deleting coarsening from the main SEMR loop entirely, but I thought it would be likely that users might want to explore it as an option so opted to keep it.

Reply:
I see that you already updated this in the last commit, but two comments:

1. If it would serve better to rename the top level `Refinement` to something else, I could be on board with that. `config["Model"]["Adaptation"]` or `"MeshAdaptation"` could work too, and the existing `"UniformLevels"` and other parameters wouldn't seem totally out of place in that section. I also do not think it is necessarily confusing to users to have `config["Refinement"]["Tol"]` or `["MaxIts"]` (as currently, but without the `"Adapt"` prefix). There shouldn't be any conflict with any other tolerances associated with refinement. But if you feel it is necessary I guess we can keep it.

2. The main benefit to moving items to the advanced section is not really avoiding doc strings (they should have probably all have doc strings but this makes them harder to change quickly as the code changes, and so for now they are undocumented). It is to be clear to users what items you intend for them to specify and which ones they really never should unless doing development work. So, items in the advanced section are not included in the demo config file above the docstrings.